sdk/orm-django/django_encryption/fields.py

```diff
@@ -42,8 +44,163 @@ class _RaiseError:
+            else:  # EncryptedMixin
+                field_name = field.name
+            clone._transform_fields[field_name] = transformation_name
+        clone._iterable_class = make_iterable_wrapper(clone._transform_fields)
+        return clone
+
+    def mask(self, *fields):
+        return self.transform(MASK_TRANSFORMATION_NAME, *fields)
+
+
+class EncryptedBatchManager(django.db.models.Manager):
+    def mask(self, *fields):
+        return self.get_queryset().mask(*fields)
+
+    def transform(self, transformation_name, *fields):
+        return self.get_queryset().transform(transformation_name, *fields)
+
+    def get_queryset(self):
+        qs = EncryptionBatchQuerySet(self.model, using=self._db)
+        for field in self.model._meta.get_fields():
+            if isinstance(field, EncryptedMixin) and field.eager:
+                qs = qs.prefetch_related(field.name)
+        return qs
+
+
+class EncryptingModel(django.db.models.Model):
+    """Base class for models that use encrypted fields,
+    allows for bult decryption and transformations as part of the queryset"""
+
+    objects = EncryptedBatchManager()
+
+    class Meta:
+        abstract = True
+
+
+# EncryptedMixinDescriptor is a descriptor wrapping access to fields inheriting from EncryptedMixin
+# it allows us to:
+#   lazily decrypt the field value when it is accessed, and then caching the result
+#   bulk decrypt values
+class EncryptedMixinDescriptor(DeferredAttribute):
+
+    def __init__(self, field):
+        super().__init__(field)
+
+    def __get__(self, instance, owner):
+        if instance is None:
+            return DeferredAttribute.__get__(self, instance, owner)
+        if hasattr(instance, 'decrypted_' + self.field.name):
+            decrypted_value = getattr(instance, 'decrypted_' + self.field.name)
+            return decrypted_value
+        if hasattr(instance, 'encrypted_' + self.field.name):
+            encrypted_value = getattr(instance, 'encrypted_' + self.field.name)
+            if encrypted_value is None:
+                setattr(instance, 'decrypted_' + self.field.name, None)
+                return None
+
+            transformation = None
+            if hasattr(instance, '_transform_fields'):
+                transformation = instance._transform_fields.get(
+                    self.field.name)
+            decrypted_value = self.field.get_decrypted_value(
+                encrypted_value, transformation=transformation)
+            setattr(instance, 'decrypted_' + self.field.name, decrypted_value)
+            return decrypted_value
+        return DeferredAttribute.__get__(self, instance, owner)
+
+    def __set__(self, instance, value):
+        if instance is None:
+            raise AttributeError("Can only be set on instances")
+        if value is None:
+            setattr(instance, 'decrypted_' + self.field.name, None)
+            setattr(instance, 'encrypted_' + self.field.name, None)
```

Comment:
why twice ?

Reply:
One is encrypted value, one is decrypted value, both are None